tests/std/tests/P0355R7_calendars_and_time_zones_dates/test.cpp

```diff
@@ -0,0 +1,930 @@
+    static_assert(noexcept(wdi1.index()));
+    static_assert(noexcept(wdi1.ok()));
+
+    static_assert(noexcept(wdi1 == wdi1));
+
+    assert(wdi1.weekday() == Monday);
+    assert(wdi1.index() == 2);
+
+    weekday_indexed wdi2 = Monday[2];
+    assert(wdi2.weekday() == Monday);
+    assert(wdi2.index() == 2);
+
+    assert(wdi1 == wdi2);
+
+    assert(!weekday_indexed(Sunday, 0).ok());
+    for (unsigned int i = 1; i <= 5; ++i) {
+        assert(weekday_indexed(Sunday, i).ok());
+        assert(weekday_indexed(Monday, i).ok());
+        assert(weekday_indexed(Tuesday, i).ok());
+        assert(weekday_indexed(Wednesday, i).ok());
+        assert(weekday_indexed(Thursday, i).ok());
+        assert(weekday_indexed(Friday, i).ok());
+        assert(weekday_indexed(Saturday, i).ok());
+    }
+    assert(!weekday_indexed(Sunday, 6).ok());
+    assert(!weekday_indexed(Sunday, 7).ok());
+}
+
+constexpr void weekday_last_test() {
+    const weekday_last wdl{Monday};
+
+    static_assert(noexcept(weekday_last{weekday{}}));
+
+    static_assert(noexcept(wdl.weekday()));
+    static_assert(noexcept(wdl.ok()));
+
+    static_assert(noexcept(wdl == wdl));
+
+    assert(wdl.ok());
+    assert(Monday[last].ok());
+    assert(Monday[last].weekday() == Monday);
+    assert(wdl.weekday() == Monday);
+    assert(wdl == weekday_last{Monday});
+}
+
+constexpr void month_day_test() {
+    const month_day md{January, 1d};
+
+    static_assert(noexcept(month_day{}));
+    static_assert(noexcept(month_day{month{}, day{}}));
+
+    static_assert(noexcept(md.month()));
+    static_assert(noexcept(md.day()));
+    static_assert(noexcept(md.ok()));
+
+    static_assert(noexcept(md == md));
+    static_assert(noexcept(md <=> md));
+
+    assert(md.month() == January);
+    assert(md.day() == 1d);
+
+    assert(md < month_day(January, 2d));
+    assert(month_day(January, 2d) > md);
+    assert(md < month_day(December, 25d));
+    assert(month_day(December, 25d) > md);
+    assert(md == month_day(January, 1d));
+
+    if (is_constant_evaluated()) {
+        static_assert((January / 31).ok());
+        static_assert((February / 29).ok());
+        static_assert((April / 30).ok());
+        static_assert(!(January / 32).ok());
+        static_assert(!(February / 30).ok());
+        static_assert(!(April / 31).ok());
+    } else {
+        for (unsigned int i = 0; i <= 255; ++i) {
+            month m{i};
+            for (unsigned int d = 0; d <= 255; ++d) {
+                if (d < 1 || d > 31 || i < 1 || i > 12) {
+                    assert(!month_day(m, day{d}).ok());
+                } else if (d == 30 && m == February) {
+                    assert(!month_day(m, day{d}).ok());
+                    break;
+                } else if (d == 31 && (m == April || m == June || m == September || m == November)) {
+                    assert(!month_day(m, day{d}).ok());
+                } else if (d == 32) {
+                    assert(!month_day(m, day{d}).ok());

```

Comment:
This can't activate (and should be dropped) because `d > 31` absorbs this above.